src/app/public/shared/input/input.component.ts

```diff
@@ -1,20 +1,18 @@
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, Input } from '@angular/core';
 import { FormControl, FormGroup } from '@angular/forms';
 
 @Component({
   selector: 'app-input',
   templateUrl: './input.component.html',
   styleUrls: ['./input.component.css'],
 })
-export class InputComponent implements OnInit {
+export class InputComponent {
   constructor() {}
   @Input() inputId = '';
   @Input() control = new FormControl();
   @Input() label = '';
 
-  ngOnInit(): void {}
-
   formGroup = new FormGroup({
```

Comment:
the declaration of formgroups should go on constructor or ngoninit, doing some logic in the class domain is risky